src/main/java/org/sudoku/Cell.java

```diff
@@ -1,99 +1,63 @@
 package org.sudoku;
 
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
-
-public class Cell implements Comparable {
-
-  private Set<Lane> belongToLanes = new HashSet<>(3);
-  private Set<Byte> possibleNumbers = new HashSet<>(9);
-  private Byte value = null;
-  private byte positionX;
-  private byte positionY;
-
-  public static final Byte[] numbers = {1,2,3,4,5,6,7,8,9};
-
-  public Cell(byte positionX, byte positionY) {
-    this.positionX = positionX;
-    this.positionY = positionY;
-    this.possibleNumbers.addAll(Arrays.asList(numbers));
-  }
-
-
-  public void setBelongToLanes(Lane vLane, Lane hLane, Lane square) {
-    this.belongToLanes.add(vLane);
-    this.belongToLanes.add(hLane);
-    this.belongToLanes.add(square);
-  }
-
-  public Set<Byte> getPossibleNumbers() {
-    return possibleNumbers;
-  }
-
-  public Set<Cell> setNumberValue(byte value) {
-    if(this.value!=null) {
-      throw new SudokuException("Tried to set \""+value+"\" for "+this.debug());
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cell {
+    private int y;
+    private int x;
+    private int value;
+    public List<Integer> possibleValues;
+    private byte square;
+    public byte id;
+
+    public Cell(int x, int y){
+        this.x = x;
+        this.y = y;
+        this.id = (byte)(x - 1 + (y - 1) * 9);
+        this.value = 0;
+        this.square = (byte)(3*((y-1)/3) + (x-1)/3);
+        this.possibleValues = new ArrayList<Integer>();
     }
-    this.possibleNumbers.clear();
-    this.value = value;
-    System.out.println(debug());
 
-    Set<Cell> cells = new HashSet<>();
-
-    for(Lane lane: belongToLanes) {
-      cells.addAll(lane.collectChangedCells(value));
+    public void setCell(int x, int y, int value){
+        this.x = x;
+        this.y = y;
+        this.id = (byte)(x - 1 + (y - 1) * 9);
+        this.value = value;
+        this.square = (byte)(3*((y-1)/3) + (x-1)/3);
+        this.possibleValues = new ArrayList<Integer>();
     }
-    return cells;
-  }
 
-  public boolean removePossibleNumber(byte value) {
-    if(this.value!=null) {
-      return false;
+    public int getX(){
+        return this.x;
     }
-//    System.out.println("cell["+positionX+"]["+positionY+"] = " + this +" remove "+value);
-    return possibleNumbers.remove(value);
-  }
-
-  public boolean contains(byte value) {
-    return possibleNumbers.contains(value);
-  }
 
-  public int possibleNumbersSize() {
-    return possibleNumbers.size();
-  }
+    public int getY(){
+        return this.y;
+    }
 
-  @Override
-  public int compareTo(Object o) {
-    if (value!=null) return -1;
-    Cell cell = (Cell) o;
-    return cell.possibleNumbersSize() - this.possibleNumbersSize();
-  }
+    public int getVal(){
```

Comment:
getters and setters names should match each other and also the field name

getValue()